framework/data_types/varying.h

```diff
@@ -151,7 +151,7 @@ class Varying
       }
     }
 
-    bool operator!=(const VaryingType& that) const override { return not(*this == that); }
+    bool operator!=(const VaryingType& that) const override { return *this != that; }
```

Comment:
clang reports this warning here:
```
warning: all paths through this function will call itself [-Winfinite-recursion]
    bool operator!=(const VaryingType& that) const override { return *this != that; }
```
that's why the original code uses `==` and negation.